### PR TITLE
Bump kubevirtci 2104200713-f58c002

### PR DESCRIPTION
**Pull request by ormergi**
**What this PR does / why we need it**:
Bump kubevirtci to 2104200713-f58c002 tag, includes:

* f58c002 (tag: 2104200713-f58c002) Add support for SRIOV Live Migration on SRIOV provider (kubevirt/kubevirtci#570)

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
Currently SRIOV migration feature is not tested on CI, we already had to send patches to fix it due to PR's that broke it.
This PR enables Kubevirt to test SRIOV migration by creating a setup that satisfy the test requirements (two worker nodes with SRIOV PF's).
It will prevent the feature from breaking again.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by EdDev** [comment]
Seems to run and pass: https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/5486/pull-kubevirt-e2e-kind-1.17-sriov/1384445164138795008#1:build-log.txt%3A5056

**Review by EdDev — APPROVED** [review]

File: automation/test.sh
Comment (oshoval) [inline automation/test.sh:65]: we still have 3 nodes so it will fail 
as we see here
https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/5486/pull-kubevirt-e2e-kind-1.17-sriov/1384775716037464064

**Review by oshoval — CHANGES_REQUESTED** [review]

File: automation/test.sh
Comment (EdDev) [reply automation/test.sh:65]: What are you talking about? It needs 3 nodes for the migration to be possible.
What are you nacking here exactly and why?

Comment (ormergi) [reply automation/test.sh:65]: I added this commit (increasing  KUBEVIRT_NUM_NODES) in order to enable testing sriov migration on CI

This PR passed CI few times but did not merged do to how tide works
SRIOV migration test passes on a local setup consistently, its also stable on CI, I think its the first time it failed.
I wouldn't consider it a flake.
I am investigating it on https://github.com/kubevirt/kubevirt/pull/5238

Comment (oshoval) [reply automation/test.sh:65]: This PR adds support for migration
We see migration test is flaky
When i first saw it i thought this PR still want to have 2 nodes, so migration won't run because there were the bugs Or found.
But then understood (correct me if i am wrong), that its intentional that there will be migration which should work and be stable.
So I still left the request changes because, and edited the message, because the test is flaky, 
and its due to this PR which cause regression on the master status (adds a flake)

**Comment by oshoval** [comment]
Should we add a sanity check that if we run on CI (only), it must test migration (aka there must be 3 nodes) ?
because if there will be only two nodes it will skip it silently,
or its not needed?

**Comment by ormergi** [comment]
> Should we add a sanity check that if we run on CI (only), it must test migration (aka there must be 3 nodes) ?
> because if there will be only two nodes it will skip it silently,
> or its not needed?

I think removing the skip if there are not enough sriov workers should be good enough

**Comment by oshoval** [comment]
> > Should we add a sanity check that if we run on CI (only), it must test migration (aka there must be 3 nodes) ?
> > because if there will be only two nodes it will skip it silently,
> > or its not needed?
> 
> I think removing the skip if there are not enough sriov workers should be good enough

In this case what about Kubevirtci ?
Its better that it will test 2 workers as well,
and if so, lets add an assert that there are at least 2 workers, unless a special flag is used to override it,
(for example when developing locally, using multi sriov locally, or if just debugging with only one PF)

**Comment by ormergi** [comment]
> In this case what about Kubevirtci ?
> Its better that it will test 2 workers as well

I agree

> and if so, lets add an assert that there are at least 2 workers, unless a special flag is used to override it,
> (for example when developing locally, using multi sriov locally, or if just debugging with only one PF)

I think we should discuss how to do it, anyway we better do it on another PR
we already had regressions because its not tested on CI

**Comment by oshoval** [comment]
> I think we should discuss how to do it, anyway we better do it on another PR
> we already had regressions because its not tested on CI

Np, just lets understand those failures we see and thats it

**Comment by ormergi** [comment]
> > I think we should discuss how to do it, anyway we better do it on another PR
> > we already had regressions because its not tested on CI
> 
> Np, just lets understand those failures we see please and thats it

The failure we saw was due to a timeout on asserting thta the VF is visible inside the guest VM,
according to virt-handler, virt-launcher and the nodes dmesg logs the VF is connected back to the guest VM successfully.

I am running SRIOV lane few more times.

**Review by EdDev — APPROVED** [review]
All seems correct, we just need to see stability in the tests.
I would suggest 5 consecutive passes on the sriov job before we take this in.

Please record the last run before starting a new one.

/hold

**Comment by ormergi** [comment]
> All seems correct, we just need to see stability in the tests.
> I would suggest 5 consecutive passes on the sriov job before we take this in.
> 
> Please record the last run before starting a new one.
> 
> /hold

We can also follow it with prow PR history 
https://prow.apps.ovirt.org/pr-history/?org=kubevirt&repo=kubevirt&pr=5486

**Comment by EdDev** [comment]
```
{"component":"tests","kind":"VirtualMachineInstance","level":"info","msg":"[{1 \r\n[root@testvmi-hxghp fedora]#  [\r\n[root@testvmi-hxghp fedora]#  \r\n[root@testvmi-hxghp fedora] # ]} {3 ip link show eth0\r\nDevice \"eth0\" does not exist.\r\n[root@testvmi-hxghp fedora]#  []}]","name":"testvmi-hxghp","namespace":"kubevirt-test-default1","pos":"console.go:99","timestamp":"2021-04-22T04:44:32.277454Z","uid":"279ed4a2-bba9-4b79-807b-327583b14fe7"}
```
I think this is happening only when the SR-IOV device is loaded as `eth0` (which is not often) originally and then when plugged back it may have the name changed.
I'm going to propose a fix to the test, as it assumes the same name is preserved, but with the ethX naming pattern, that cannot be assured. Later on we should adjust the Fedora image to use the persisted naming convention for interfaces.
We also need to make sure that when migrating, the agent info for the interfaces is reset (I have not confirmed it).

**Comment by oshoval** [comment]
> I think this is happening only when the SR-IOV device is loaded as `eth0` originally (which is not often) and then when plugged back it may have the name changed.
> I'm going to propose a fix to the test, as it assumes the same name is preserved, but with the ethX naming pattern, that cannot be assured. Later on we should adjust the Fedora image to use the persisted naming convention for interfaces.
> We also need to make sure that when migrating, the agent info for the interfaces is reset (I have not confirmed it).

Would cloud-init `set-name` feature help here ?

**Comment by ormergi** [comment]
#5505 is verified with this PR changes, tested on https://github.com/kubevirt/kubevirt/pull/5238
Passed SRIOV lane 5 times
Once #5505 is merged we can continue with this one

**Comment by EdDev** [comment]
#5505 is merged.
/unhold

**Pull request merged**
